File: backend/src/main/java/net/zorphy/backend/site/catan/dto/game/GameStats.java
package net.zorphy.backend.site.catan.dto.game;

import net.zorphy.backend.main.dto.game.stats.GameSpecificStats;
import net.zorphy.backend.main.dto.game.stats.metrics.GameStatsMetrics;
import net.zorphy.backend.site.catan.dto.DiceRoll;

import java.time.Duration;
import java.util.List;


public record GameStats(
        int gameCount,
        double luckMetric,
        GameStatsMetrics<Duration> rollDurationMetrics,
        List<DiceRoll> diceRolls
) implements GameSpecificStats {
}

File: frontend/src/app/sites/catan/dto/game/GameStats.ts
import {DiceRoll} from "../DiceRoll";
import {LinkedGameStats} from "../../../../main/dto/games/stats/LinkedGameStats";

export interface GameStats {
    gameCount: number;
    luckMetric: number;
    minRollDuration: LinkedGameStats<string>,
    maxRollDuration: LinkedGameStats<string>,
    avgRollDuration: string,
    diceRolls: DiceRoll[]
}